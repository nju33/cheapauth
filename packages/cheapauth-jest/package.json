{
  "private": true,
  "name": "cheapauth-jest",
  "version": "0.0.2-alpha.0",
  "dependencies": {
    "@testing-library/react": "^8.0.1",
    "@types/jest": "26.0.15",
    "@vue/test-utils": "^1.0.0-beta.29",
    "jest": "24.9.0",
    "react-testing-library": "^8.0.1"
  },
  "devDependencies": {
    "typescript": "3.9.7",
    "vue": "^2.6.10"
  },
  "peerDependencies": {
    "vue": "^2.6.0"
  }
}
