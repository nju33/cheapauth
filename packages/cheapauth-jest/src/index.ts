import {RenderResult} from '@testing-library/react';
import Vue from 'vue';
import {Wrapper} from '@vue/test-utils';
import TestId from './constants/testid';

export type CheapauthRenderResult = RenderResult | Wrapper<Vue>;

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toBeTitleTextContent(text: string): R;
      toBeSubmitTextContent(text: string): R;
      toBeErrorMessageTextContent(text: string): R;
      toBeApproved(): R;
    }
  }
}

const isReactTest = (
  renderResult: CheapauthRenderResult,
): renderResult is RenderResult => {
  return Object.prototype.hasOwnProperty.call(renderResult, 'asFragment');
};

const pass = {
  pass: true,
  message: '',
};

const toBeOneTextContent = (testid: TestId) => (
  renderResult: CheapauthRenderResult,
  text: string,
): jest.CustomMatcherResult => {
  if (isReactTest(renderResult)) {
    const {queryByTestId} = renderResult;

    const targetElement = queryByTestId(testid);
    if (targetElement === null) {
      return {
        pass: false,
        message: () => {
          return `expected to have the element \`[data-testid='${testid}']\``;
        },
      };
    }

    if (targetElement.textContent !== text) {
      return {
        pass: false,
        message: () => {
          return `expected to have the element \`[data-testid='${testid}']{${
            targetElement.textContent
          }}\``;
        },
      };
    }
  } else {
    const titleWrapper = renderResult.find(`[data-testid='${testid}']`);
    const expectedText = titleWrapper.text();

    if (!(titleWrapper.exists() && expectedText === text)) {
      return {
        pass: false,
        message: () => {
          return `expected to have the element \`[data-testid='${testid}']{${expectedText}}\``;
        },
      };
    }
  }

  return pass;
};

expect.extend({
  toBeTitleTextContent: toBeOneTextContent(TestId.title),
  toBeSubmitTextContent: toBeOneTextContent(TestId.submit),
  toBeErrorMessageTextContent: toBeOneTextContent(TestId.errorMessage),
  toBeApproved(renderResult: CheapauthRenderResult): jest.CustomMatcherResult {
    if (isReactTest(renderResult)) {
      const targetElement = renderResult.queryByTestId(TestId.root);
      if (targetElement === null) {
        return {
          pass: false,
          message: () => {
            return `expected to have to element \`[data-testid='${
              TestId.root
            }']\``;
          },
        };
      }

      if (targetElement.dataset.approval !== 'true') {
        return {
          pass: false,
          message: () => {
            return `expected to have to element \`[data-testid='${
              TestId.root
            }'][data-approval='true']\``;
          },
        };
      }
    } else {
      const titleWrapper = renderResult.find(`[data-testid='${TestId.root}']`);
      const approval = titleWrapper.attributes()['data-approval'];

      if (!(titleWrapper.exists() && approval === 'true')) {
        return {
          pass: false,
          message: () => {
            return `expected to have to element \`[data-testid='${
              TestId.root
            }'][data-approval='true']\``;
          },
        };
      }
    }

    return pass;
  },
});
