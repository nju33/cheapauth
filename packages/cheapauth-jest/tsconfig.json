{
  "compilerOptions": {
    "target": "es5",
    "module": "esnext",
    "typeRoots": ["./node_modules/@types", "../../node_modules/@types"],
    "jsx": "react",
    "sourceMap": true,
    "strict": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "downlevelIteration": true
  },
  "include": ["src"],
  "exclude": ["node_modules"]
}
