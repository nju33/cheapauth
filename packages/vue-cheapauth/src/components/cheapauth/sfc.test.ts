import Vue from 'vue';
import {shallowMount, Wrapper} from '@vue/test-utils';
// @ts-ignore
import Cheapauth from './sfc/index.vue';
import 'cheapauth-jest/extend-expect';

describe('Cheapauth', () => {
  const password = 'password';
  const title = 'タイトル';
  const submitLabel = 'ログインする';

  let wrapper: Wrapper<Vue>;

  beforeEach(() => {
    wrapper = shallowMount(Cheapauth, {
      propsData: {
        password,
        title,
        submitLabel,
      },
    });
  });

  it('is rendered with text of props into right place', () => {
    expect(wrapper).toBeTitleTextContent(title);
    expect(wrapper).toBeSubmitTextContent(submitLabel);
  });

  it('is input password is incorrect', () => {
    const incorrectPassword = 'passworddd';

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper).not.toBeApproved();

    // to fire the submit event
    wrapper.setData({input: incorrectPassword});
    wrapper.find(`[data-testid='cheapauth--form']`).trigger('submit');

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper).toBeErrorMessageTextContent('パスワードが違います');
    expect(wrapper).not.toBeApproved();
  });

  it('is input password is correct', () => {
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper).not.toBeApproved();

    // to fire the submit event
    wrapper.setData({input: password});
    wrapper.find(`[data-testid='cheapauth--form']`).trigger('submit');

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper).toBeApproved();
  });
});
