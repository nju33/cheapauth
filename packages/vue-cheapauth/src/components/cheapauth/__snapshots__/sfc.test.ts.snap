// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheapauth is input password is correct 1`] = `
<div
  class="cheapauth--root"
  data-testid="cheapauth--root"
>
  <div
    class="cheapauth--document"
  >
    <section
      aria-labelledby="cheapauth"
      class="cheapauth--section"
    >
      <header
        class="cheapauth--header"
      >
        <h3
          class="cheapauth--title"
          data-testid="cheapauth--title"
        >
          タイトル
        </h3>
      </header>
       
      <div>
        <form
          class="cheapauth--form"
          data-testid="cheapauth--form"
        >
          <div
            class="cheapauth--row"
          >
            <input
              class="cheapauth--input"
              data-testid="cheapauth--input"
              type="password"
            />
             
            <!---->
          </div>
           
          <div
            class="cheapauth--row"
          >
            <button
              class="cheapauth--submit"
              data-testid="cheapauth--submit"
            >
              ログインする
            </button>
          </div>
        </form>
      </div>
    </section>
  </div>
   
  <!---->
</div>
`;

exports[`Cheapauth is input password is correct 2`] = `
<div
  class="cheapauth--root"
  data-approval="true"
  data-testid="cheapauth--root"
>
  <div
    class="cheapauth--document"
  >
    <section
      aria-labelledby="cheapauth"
      class="cheapauth--section"
    >
      <header
        class="cheapauth--header"
      >
        <h3
          class="cheapauth--title"
          data-testid="cheapauth--title"
        >
          タイトル
        </h3>
      </header>
       
      <div>
        <form
          class="cheapauth--form"
          data-testid="cheapauth--form"
        >
          <div
            class="cheapauth--row"
          >
            <input
              class="cheapauth--input"
              data-testid="cheapauth--input"
              type="password"
            />
             
            <!---->
          </div>
           
          <div
            class="cheapauth--row"
          >
            <button
              class="cheapauth--submit"
              data-testid="cheapauth--submit"
            >
              ログインする
            </button>
          </div>
        </form>
      </div>
    </section>
  </div>
   
</div>
`;

exports[`Cheapauth is input password is incorrect 1`] = `
<div
  class="cheapauth--root"
  data-testid="cheapauth--root"
>
  <div
    class="cheapauth--document"
  >
    <section
      aria-labelledby="cheapauth"
      class="cheapauth--section"
    >
      <header
        class="cheapauth--header"
      >
        <h3
          class="cheapauth--title"
          data-testid="cheapauth--title"
        >
          タイトル
        </h3>
      </header>
       
      <div>
        <form
          class="cheapauth--form"
          data-testid="cheapauth--form"
        >
          <div
            class="cheapauth--row"
          >
            <input
              class="cheapauth--input"
              data-testid="cheapauth--input"
              type="password"
            />
             
            <!---->
          </div>
           
          <div
            class="cheapauth--row"
          >
            <button
              class="cheapauth--submit"
              data-testid="cheapauth--submit"
            >
              ログインする
            </button>
          </div>
        </form>
      </div>
    </section>
  </div>
   
  <!---->
</div>
`;

exports[`Cheapauth is input password is incorrect 2`] = `
<div
  class="cheapauth--root"
  data-testid="cheapauth--root"
>
  <div
    class="cheapauth--document"
  >
    <section
      aria-labelledby="cheapauth"
      class="cheapauth--section"
    >
      <header
        class="cheapauth--header"
      >
        <h3
          class="cheapauth--title"
          data-testid="cheapauth--title"
        >
          タイトル
        </h3>
      </header>
       
      <div>
        <form
          class="cheapauth--form"
          data-testid="cheapauth--form"
        >
          <div
            class="cheapauth--row"
          >
            <input
              class="cheapauth--input"
              data-testid="cheapauth--input"
              type="password"
            />
             
            <div
              aria-live="assertive"
              class="cheapauth--error-message"
              data-testid="cheapauth--error-message"
            >
              <small>
                パスワードが違います
              </small>
            </div>
          </div>
           
          <div
            class="cheapauth--row"
          >
            <button
              class="cheapauth--submit"
              data-testid="cheapauth--submit"
            >
              ログインする
            </button>
          </div>
        </form>
      </div>
    </section>
  </div>
   
  <!---->
</div>
`;
